### PR TITLE
Share amount and tokens

### PR DESCRIPTION
**Pull request by zoek1**
##### Description
- Share cart with amount and token included
- Persist comments when the user refresh
https://www.loom.com/share/092a487867f44753a6e56c43c80a00d2

### CODE REVIEW

File: app/grants/views.py [hunk 1]
```diff
@@ -922,26 +923,47 @@ def grants_cart_view(request):
...
     key = hashlib.md5(grant_str.encode('utf')).hexdigest()
     views = redis.incr(key)
 
-    grant_ids = grant_str.split(':')[0].split(',')
-    grant_ids = [int(ele) for ele in grant_ids if ele and ele.isnumeric() ]
+    grants_data = grant_str.split(':')[0].split(',')
+
+    for ele in grants_data:
+        # new format will support amount and token in the URL separated by ;
+        grant_data = ele.split(';')
```
Comment (owocki) [inline app/grants/views.py:935]: is this backwards compatible with the old links that have already been shared out in the wild?

**Comment by owocki** [comment]
https://bits.owocki.com/RBuq96nd

hmm it isnt working for me for some reason. i had 2 DAI grants and 1 ETH grant in my cart

**Comment by owocki** [comment]
hmmm yeah it just seems to clear my cart.  derp 

https://gitcoin.co/grants/cart/bulk-add/351;1;283,237;5;283,15;1;283,13;5;283,:owocki

**Comment by zoek1** [comment]
The current "share My cart" prefills automatically the domain with gitcoin.co, it needs substitute manually the domain of your host in testing

**Comment by owocki** [comment]
just tested this and it works fine. looks like its backwards compartible with the old style links too ( http://localhost:8000/grants/cart/bulk-add/567,143,:owocki ) cool!

**Comment by owocki** [comment]
:shipit:

**Pull request merged**